### DISCUSSION TITLE
Get permissions repeatedly in macos

### DISCUSSION THREAD

**Discussion by Liko209** [tauri-apps/tauri#10779] [Q&A]
Every time after I update my apps via updater, the system requires me to regain permissions, for example, I need screen recording permissions in my apps. But when I open the settings it shows that the app's permissions are turned on, and it doesn't work when I turn it off and on again. I had to reset my app's permissions first with the "tccutil reset ScreenCapture xxx" command, then open the settings page to add the app's permissions.

I asked gpt and it told me to look at my app's info via the codesign -dvvv command and check that the identifiers in there are consistent. I found that even though I configured the identifier in tauri.config.json, every time I build and use codesign -dvvv to look at the identifier of my app, it is different.

So I have two questions.
One, how do I configure it to make sure I don't need to reset my app permissions after every update.
Two, why is my identifier randomized every time?

Has anyone had a similar issue?

**Comment by FabianLars** [comment]
As far as i know you must codesign your app for permissions to persist across app versions / to keep the identifier consistent.
